StyleCop.Analyzers/StyleCop.Analyzers/SpacingRules/SA1009ClosingParenthesisMustBeSpacedCorrectly.cs

```diff
@@ -204,6 +204,13 @@ private static void HandleCloseParenToken(SyntaxTreeAnalysisContext context, Syn
                 precedesStickyCharacter = true;
                 break;
 
+            case SyntaxKind.IfStatement:
+            case SyntaxKind.WhileStatement:
+            case SyntaxKind.ForStatement:
+            case SyntaxKind.ForEachStatement:
+                precedesStickyCharacter = false;
+                break;
```

Comment:
💭 I think I would prefer this to be relocated to the handling of `PlusPlusToken`/`MinusMinusToken` above, and the new implementation in that section should be similar to the way `PlusToken` and `MinusToken` and handled further above.

Reply:
Updated. Did you have something like that in mind?